### PR TITLE
[Kogito-8520] Separate plain builder API and Builder-k8s API

### PR DESCRIPTION
**Pull request by desmax74**
WIP Merge after https://github.com/kiegroup/container-builder/pull/18

See: https://issues.redhat.com/browse/KOGITO-8520
Signed-off-by: max <mdessi@redhat.com>

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -172,11 +175,13 @@ func (b *Builder) ScheduleBuild(kb KogitoBuilder) (*api.Build, error) {
 			},
 		},
 	}
-
-	build, err := builder.NewBuild(platform, kb.ImageName, kb.PodMiddleName).
-		WithKanikoCache(kb.Cache).
-		WithKanikoResources(kb.Resources).
-		WithKanikoAdditionalArgs(kb.AdditionalFlags).
+	infoBuild := builder.BuilderInfo{Platform: platform,
```
Comment (spolti) [inline builder/builder.go:178]: instead replicate can't we have a common helper func for it?
the build, err := part seems to be equal too.

File: go.mod [hunk 2]
```diff
@@ -104,3 +185,6 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+// remove when the https://github.com/kiegroup/container-builder/pull/18 will be merged
```
Comment (spolti) [inline go.mod:189]: all those dependency changes were introduced only by the builder version change?

Comment (desmax74) [reply go.mod]: fixed

File: builder/builder.go (hunk 1 above)
Comment (desmax74) [reply builder/builder.go]: added

**Review by davidesalerno — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**